### ISSUE TITLE
Also auto-jump to the right position in markdown-preview(-plus) when clicking document outline [feature request]

### ISSUE THREAD

**Issue by salim-b** [mangecoeur/document-outline#38]
Right now when I click on a header in the document outline, the editor jumps automatically to the coresponding position in the source file. When I write Markdown, I normally have a preview opened side-by-side with the source (using package `markdown-preview-plus`). Therefore it would be awesome to have the same auto-jump functionality for the preview, too!

**Comment by awmartin** [comment]
How does one enable this click-to-jump feature. When I click on a header in the document outline, nothing happens.

**Comment by awmartin** [comment]
I see. It just scrolls, unless you are already there. I would expect it to focus the editor and place the cursor on that header line.

**Comment by mangecoeur** [comment]
@awmartin it doesn't move the cursor, under the idea that you might want to jump to a heading to look up something without loosing the place where you were typing.

**Comment by awmartin** [comment]
@mangecoeur Yeah, I understand, but that doesn't really match my workflow. I want to click to navigate, not just look things up and go back, which is much rarer for me and involves too much clicking. This is also consistent with other editors I use, like Google Docs. I also generally remember where I'm coming from, and use bookmarks to hold my place.

But that's ok, I've already modified my fork to do what I need (move the cursor, select the header in the editor, and focus the editor pane). Sounds like a behavior that could be an option though, if you're open to PRs.

**Comment by salim-b** [comment]
> it doesn't move the cursor, under the idea that you might want to jump to a heading to look up something without loosing the place where you were typing.

Generally that seems sensible. But there's a problem: By clicking an entry in the document outline the cursor currently loses focus. This makes it kinda hard to figure out where you came from (at least in large documents; the only hint seems the highlighting in the document outline). Is there a shortcut to get back to your cursor position or something?

>  I've already modified my fork to do what I need (move the cursor, select the header in the editor, and focus the editor pane). Sounds like a behavior that could be **an option** though

:+1:

**Comment by mangecoeur** [comment]
> by clicking an entry in the document outline the cursor currently loses focus

that's a bit atom's fault, this didn't happen with the old 'panels' system but the new 'docks' messes with the focus. Not really dug enough into this. If you just start typing it should jump back to where you were before (though TBH not sure if that detail actually works right now).

**Comment by salim-b** [comment]
> If you just start typing it should jump back to where you were before (though TBH not sure if that detail actually works right now).

Nope, doesn't work. Nothing happens when you start typing ('cause the cursor doesn't have focus 😜  ).

**Comment by mangecoeur** [comment]
@salim-b bum, will open separate issue for that.
